src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -306,47 +272,45 @@ private bool TryBindUnconvertedBinaryOperatorToDefaultInterpolatedStringHandler(
                 additionalConstructorRefKinds: default);
 
             // Now that the parts have been bound, reconstruct the binary operators.
-            convertedBinaryOperator = UpdateBinaryOperatorWithInterpolatedContents(stack, appendCalls, data, binaryOperator.Syntax, diagnostics);
-            stack.Free();
+            convertedBinaryOperator = UpdateBinaryOperatorWithInterpolatedContents(binaryOperator, appendCalls, data, binaryOperator.Syntax, diagnostics);
             return true;
         }
 
-        private BoundBinaryOperator UpdateBinaryOperatorWithInterpolatedContents(ArrayBuilder<BoundBinaryOperator> stack, ImmutableArray<ImmutableArray<BoundExpression>> appendCalls, InterpolatedStringHandlerData data, SyntaxNode rootSyntax, BindingDiagnosticBag diagnostics)
+        private BoundBinaryOperator UpdateBinaryOperatorWithInterpolatedContents(BoundBinaryOperator originalOperator, ImmutableArray<ImmutableArray<BoundExpression>> appendCalls, InterpolatedStringHandlerData data, SyntaxNode rootSyntax, BindingDiagnosticBag diagnostics)
         {
-            Debug.Assert(appendCalls.Length == stack.Count + 1);
             var @string = GetSpecialType(SpecialType.System_String, diagnostics, rootSyntax);
 
-            var bottomOperator = stack.Pop();
-            var result = createBinaryOperator(bottomOperator, createInterpolation(bottomOperator.Left, appendCalls[0]), rightIndex: 1);
+            Func<(ImmutableArray<ImmutableArray<BoundExpression>>, TypeSymbol), BoundUnconvertedInterpolatedString, int, BoundExpression> interpolationFactory =
```

Comment:
Are these locals useful? (here and other similar call sites)

Reply:
The types of the local functions are more strict than the delegate types themselves (BoundBinaryOperator vs BoundExpression) so I'd need to either cast or include the type parameters. This seemed the cleanest way to me.

Reply:
I didn't follow that. Feels like the local functions should just use the same signature.